zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -15,6 +15,131 @@ use crate::{
     transparent::{self, Script},
 };
 
+// Used by boilerplate code below.
+
+#[derive(Clone, Debug)]
+struct TransparentAuth<'a> {
+    all_prev_outputs: &'a [transparent::Output],
+}
+
+impl zcash_primitives::transaction::components::transparent::Authorization for TransparentAuth<'_> {
+    type ScriptSig = zcash_primitives::legacy::Script;
+}
+
+// In this block we convert our Output to a librustzcash to TxOut.
+// (We could do the serialize/deserialize route but it's simple enough to convert manually)
+impl zcash_primitives::transaction::sighash::TransparentAuthorizingContext for TransparentAuth<'_> {
+    fn input_amounts(&self) -> Vec<zcash_primitives::transaction::components::amount::Amount> {
+        self.all_prev_outputs
+            .iter()
+            .map(|prevout| {
+                zcash_primitives::transaction::components::amount::Amount::from_nonnegative_i64_le_bytes(
+                    prevout.value.to_bytes(),
+                ).expect("will not fail since it was previously validated")
+            })
+            .collect()
+    }
+
+    fn input_scriptpubkeys(&self) -> Vec<zcash_primitives::legacy::Script> {
+        self.all_prev_outputs
+            .iter()
+            .map(|prevout| {
+                zcash_primitives::legacy::Script(prevout.lock_script.as_raw_bytes().into())
+            })
+            .collect()
+    }
+}
+
+// Boilerplate mostly copied from `zcash/src/rust/src/transaction_ffi.rs` which is required
```

Comment:
I tried but it added a circular dependency (the boilerplate uses `TransparentAuth` from the module, the module uses the boilerplate) which could be solved by moving `TransparentAuth` to a new file, but at that point I think it's simpler to keep it as is since we're not likely to change this code that much in the future